chia/wallet/nft_wallet/nft_puzzles.py

```diff
@@ -83,21 +83,29 @@ def get_nft_info_from_puzzle(puzzle: Program, nft_coin: Coin) -> NFTInfo:
     data_uris = []
     for uri in uncurried_nft.data_uris.as_python():
         data_uris.append(str(uri, "utf-8"))
+    meta_uris = []
+    for uri in uncurried_nft.meta_uris.as_python():
+        meta_uris.append(str(uri, "utf-8"))
+    license_uris = []
+    for uri in uncurried_nft.license_uris.as_python():
+        license_uris.append(str(uri, "utf-8"))
 
     nft_info = NFTInfo(
-        uncurried_nft.singleton_launcher_id.as_python().hex().upper(),
+        str(uncurried_nft.singleton_launcher_id).upper(),
         nft_coin.name().hex().upper(),
-        uncurried_nft.owner_did.as_python().hex().upper(),
+        str(uncurried_nft.owner_did).upper(),
         uint64(uncurried_nft.trade_price_percentage.as_int()),
         data_uris,
-        uncurried_nft.data_hash.as_python().hex().upper(),
-        [],
-        "",
-        [],
-        "",
-        "NFT0",
-        uint64(1),
-        uint64(1),
+        str(uncurried_nft.data_hash).upper(),
+        meta_uris,
```

Comment:
@paninaro for `*_uris` fields, given that they should all be the same and only added if url is obsolete or wrong, shouldn't we be returning only the first and most recent one vs all?
returning old urls might cause security problems if NFT metadata gets compromised and NFT owner wants to effectively remove the old one

Reply:
But the metadata hash guards against that case, as modifying any of the URLs will invalidate the hash verification performed by the frontend.

I would ask that we return the first 3 or so in case the most recent addition(s) aren't available due to transient network/service issues. But sure, we probably don't need to return the entire list if it contains many entries.